pkg/controller/routes.go

```diff
@@ -64,6 +64,7 @@ func routes(router *router.Router, cfg *rest.Config, registryTransport http.Roun
 	appRouter.HandlerFunc(appdefinition.PullAppImage(registryTransport, recorder))
 	appRouter.HandlerFunc(images.CreateImages)
 	appRouter.HandlerFunc(appdefinition.ParseAppImage)
+	appRouter.HandlerFunc(appdefinition.SetImpliedAutoUpgrade)
```

Comment:
I feel like it is a little overkill to have an entire handler for this. I think it should be included in one of the other ones, like `appstatus.PrepareStatus`, or, farther down, `appstatus.SetStatus`. I'm not 100% on this though and am curious to know what others think.

Reply:
You bring up a good point. Let me look around for somewhere this may feel more idiomatic.

Reply:
Gave it some thought, what if we added an `ObservedAutoUpgrade` field and it gets set in `UpdateObservedFields`?